Signal/src/ViewControllers/MediaPageViewController.swift

```diff
@@ -609,5 +621,19 @@ class MediaPageViewController: UIPageViewController, UIPageViewControllerDataSou
         let date = Date(timeIntervalSince1970: Double(item.message.timestamp) / 1000)
         let formattedDate = dateFormatter.string(from: date)
         headerDateLabel.text = formattedDate
+
+        if #available(iOS 11, *) {
+            // Do nothing, on iOS11, autolayout grows the stack view as necessary.
+        } else {
+            // Size the titleView to be large enough to fit the widest label,
+            // but no larger. If we go for a "full width" label, our title view
+            // will not be centered (since the left and right bar buttons have different widths)            
+            headerNameLabel.sizeToFit()
+            headerDateLabel.sizeToFit()
+            let maxWidth = max(headerNameLabel.frame.width, headerDateLabel.frame.width)
+
+            let headerFrame: CGRect = CGRect(x: 0, y: 0, width: maxWidth, height: 44)
```

Comment:
I wonder if it would be better to hardcode 44 or to use `headerNameLabel.height + headerDateLabel.height`.

Reply:
That could work, but then I'm not sure I could vertically position as flexibly, e.g. note how the container margins on top are 2, but on the bottom are 4.

It's probably doable, but I found the current approach to be flexible.